bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -350,9 +311,172 @@ private static String extractName(final GuideGroup group) {
             return tmpRows;
         }
 
+        // Replace the auto group and return a new TableData.
+        // Note that we work under the assumption that there is ALWAYS an auto group, which should be the case,
+        // and that it is in row position 1 of the table (directly after the base).
+        void replaceAutoGroup(final Option<ObsContext> ctxOpt, final TargetEnvironment newEnv) {
+            final List<Row>         oldRows = rows;
+
+            // Recalculate the rows to see if quality has changed based on the new context.
+            env = newEnv;
+            final Set<Integer> changedRowIndices      = recalculateManualGSQuality(ctxOpt);
+
+            // The auto group is always in group position 0 and row position 1.
+            // Recreate it and set it in the new rows.
+            final GuideGroup autoGroup  = newEnv.getGuideEnvironment().getOptions().head();
+            final GroupRow autoGroupRow = createGroupRow(ctxOpt, 0, autoGroup);
+            rows.set(1, autoGroupRow);
+            numRows = countRows();
+
+            // Determine the size difference between the old auto group and the new.
+            final int oldAutoGroupSize  = oldRows.get(1).children().size();
+            final int newAutoGroupSize  = autoGroupRow.children().size();
+            final int autoGroupSizeDiff = newAutoGroupSize - oldAutoGroupSize;
+            final int autoGroupSizeInt  = Math.min(oldAutoGroupSize, newAutoGroupSize);
+
+            // Fire events indicating the changes in the model.
+            if (autoGroupSizeInt > 0) {
+                fireTableRowsUpdated(FIRST_AUTO_TARGET_IDX, FIRST_AUTO_TARGET_IDX + autoGroupSizeInt - 1);
+            }
+            if (oldAutoGroupSize < newAutoGroupSize) {
+                fireTableRowsInserted(FIRST_AUTO_TARGET_IDX + oldAutoGroupSize, FIRST_AUTO_TARGET_IDX + newAutoGroupSize - 1);
+            } else if (newAutoGroupSize < oldAutoGroupSize) {
+                fireTableRowsDeleted(FIRST_AUTO_TARGET_IDX + newAutoGroupSize, FIRST_AUTO_TARGET_IDX + oldAutoGroupSize - 1);
+            }
+
+            // Now for each group that was changed, fire update events.
+            int tableRowIdx = 0;
+            final Stream<Tuple2<Integer,Row>> indexedRows = IntStream.range(0, rows.size()).mapToObj(i -> new Pair<>(i, rows.get(i)));
+            for (final Tuple2<Integer, Row> tup2 : indexedRows.collect(Collectors.toList())) {
+                final int rowIdx = tup2._1();
+                final Row row    = tup2._2();
+
+                if (changedRowIndices.contains(rowIdx)) {
+                    // We only care about the child rows (hence + 1), and all rows have been offset by autoGroupSizeDiff.
+                    final int lowerIdx = tableRowIdx + 1 + autoGroupSizeDiff;
+                    final int upperIdx = lowerIdx + row.children().size() - 1;
+                    fireTableRowsUpdated(lowerIdx, upperIdx);
+                }
```

Comment:
I've tested it with GeMS quite a bit. (I hope thoroughly! I can't think of anything I might have missed.) Everything seems to work.

The reason why other rows may have changed is through the setting of the PA from the auto group. If the auto group is the primary and a BAGS lookup is triggered where another auto group with, say, a PA+180 is used, then the PA is set to 180. The guiding quality of the manual guide stars might (likely will) change as a result, as they will no longer be in the guider patrol field.

Thus, in order to set their guiding feedback correctly, it is necessary to recalculate each group and determine which, if any, have changed.

Does that make sense?